coalib/bears/Bear.py

```diff
@@ -26,6 +26,10 @@
 
 class Debugger(pdb.Pdb):
 
+    def __init__(self, *args, bear=None, **kwargs):
+        self.bear = bear
+        super(Debugger, self).__init__(*args, **kwargs)
```

Comment:
invoke `super.__init__` always as soon as possible.

Reply:
It's a good habit to not run into weird states of the object, since you don't know what might happen to `self.bear` in the meanwhile. C++ for example enforces you to initialize other supers first (a slightly different syntax though), or Java does the same, which is often a good decision.